commitizen/changelog_formats/base.py

```diff
@@ -24,10 +25,29 @@ def __init__(self, config: BaseConfig):
         # Constructor needs to be redefined because `Protocol` prevent instantiation by default
         # See: https://bugs.python.org/issue44807
         self.config = config
+        self.tag_format = self.config.settings["tag_format"]
 
     @property
     def version_parser(self) -> Pattern:
-        return get_version_scheme(self.config).parser
+        tag_regex: str = self.tag_format
+        version_regex = get_version_scheme(self.config).parser.pattern
+        TAG_FORMAT_REGEXS = {
+            "$version": version_regex,
+            "$major": r"(?P<major>\d+)",
+            "$minor": r"(?P<minor>\d+)",
+            "$patch": r"(?P<patch>\d+)",
+            "$prerelease": r"(?P<prerelease>\w+\d+)?",
+            "$devrelease": r"(?P<devrelease>\.dev\d+)?",
+            "${version}": version_regex,
+            "${major}": r"(?P<major>\d+)",
+            "${minor}": r"(?P<minor>\d+)",
+            "${patch}": r"(?P<patch>\d+)",
+            "${prerelease}": r"(?P<prerelease>\w+\d+)?",
+            "${devrelease}": r"(?P<devrelease>\.dev\d+)?",
+        }
```

Comment:
Those are exactly the same as in `changelog:get_version_tag` so I suggest to factorize this into a public factory, something like:
```python
TAG_FORMAT_REGEXS = {
    "$major": r"(?P<major>\d+)",
    "$minor": r"(?P<minor>\d+)",
    "$patch": r"(?P<patch>\d+)",
    "$prerelease": r"(?P<prerelease>\w+\d+)?",
    "$devrelease": r"(?P<devrelease>\.dev\d+)?",
    "${version}": version_regex,
    "${major}": r"(?P<major>\d+)",
    "${minor}": r"(?P<minor>\d+)",
    "${patch}": r"(?P<patch>\d+)",
    "${prerelease}": r"(?P<prerelease>\w+\d+)?",
    "${devrelease}": r"(?P<devrelease>\.dev\d+)?",
}

def tag_format_regexps_for(version: Pattern) -> dict[str, Pattern]:
    return {
        "$version": version,
		"${version}": version,
		**TAG_FORMAT_REGEXS
    }
```

This way:
- single source of thrust
- custom format implementations can reuse `TAG_FORMAT_REGEXS` and `tag_format_regexps_for`